geo/src/algorithm/rotate.rs

```diff
@@ -217,6 +218,16 @@ where
     }
 }
 
+pub trait RotateCentroid<T: CoordinateType>: RotatePoint<T> + Centroid<Output = Point<T>> {
+    fn rotate_around_centroid(&self, angle: T) -> Self
+    where
+        T: Float,
+        Self: Sized,
+    {
+        self.rotate_around_point(angle, self.centroid())
```

Comment:
I had some input which I thought would be better to discuss in the issue rather than the PR, please see here: https://github.com/georust/geo/issues/412#issuecomment-740996625

Curious about your opinion here.